src/broker/content-cache.c

```diff
@@ -858,12 +860,33 @@ static struct flux_msg_handler_spec handlers[] = {
     FLUX_MSGHANDLER_TABLE_END,
 };
 
+static flux_msg_handler_t *add_handler (flux_t *h, char *topic, uint32_t roles,
+                                        flux_msg_handler_f cb, void *arg)
+{
+    flux_msg_handler_t *w;
+    struct flux_match match = FLUX_MATCH_REQUEST;
+
+    match.topic_glob = topic;
+    if (!(w = flux_msg_handler_create (h, match, cb, arg)))
+        return NULL;
+    flux_msg_handler_allow_rolemask (w, roles);
+    flux_msg_handler_start (w);
+    return w;
+}
+
+
 int content_cache_set_flux (content_cache_t *cache, flux_t *h)
 {
     cache->h = h;
 
     if (flux_msg_handler_addvec (h, handlers, cache) < 0)
-	return -1;
+        return -1;
+    if (!(cache->load_w = add_handler (h, "content.load", FLUX_ROLE_USER,
+                                       content_load_request, cache)))
+        return -1;
+    if (!(cache->store_w = add_handler (h, "content.store", FLUX_ROLE_USER,
+                                        content_store_request, cache)))
+        return -1;
     if (flux_get_rank (h, &cache->rank) < 0)
```

Comment:
I thought leaving fields off the initializer was only possible with designated initializers, however it does appear even with traditional struct initializer you can leave off trailing members. However, I'm also not sure if these are guaranteed to be zeroed -- you're probably right that depending on implicit zeroing is unwise.

Since this is security related, it seems wise to require an explicit call to enable access to roles other than FLUX_ROLE_OWNER, instead of relying on implicit behavior, even if initialization of members is guaranteed to be zeroed.

The convenience function to find a msghandler in a msg_handler_vec is trivial, so I guess for now service authors can roll their own, though it would be nice to discourage both explicit indicies and having msghandlers split between explicit creation and a flux_msg_handler_vec.